frb_example/benches/dart/lib/interceptor.dart

```diff
@@ -0,0 +1,182 @@
+/// interceptor (platform-agnostic wrapper)
+///
+/// record time before and after executing async or sync functions over the bridge.
+///
+/// can also generate
+/// a [continuous-benchmark](https://github.com/marketplace/actions/continuous-benchmark)-compliant json output on demand
+
+import 'dart:async';
+
+import 'package:flutter_rust_bridge/flutter_rust_bridge.dart';
+import 'package:logging/logging.dart';
+import 'bridge_definitions.dart';
+import 'package:uuid/uuid.dart';
+import 'interceptor.io.dart' if (dart.library.html) 'interceptor.web.dart';
+import 'bridge_generated.io.dart' if (dart.library.html) 'bridge_generated.web.dart';
+
+/// this is the custom platform, which contains the interceptor.
+abstract class FlutterRustBridgeExampleBenchmarkSuitePlatformBenchBase<
+    STDOUT extends FlutterRustBridgeInterceptorStdOut,
+    JSON extends FlutterRustBridgeInterceptorJson> extends FlutterRustBridgeExampleBenchmarkSuitePlatform {
+  FlutterRustBridgeExampleBenchmarkSuitePlatformBenchBase(ExternalLibrary dylib) : super(dylib);
+  FlutterRustBridgeInterceptor<TimeWatch> get interceptor;
+
+  @override
+  Future<S> executeNormal<S>(FlutterRustBridgeTask<S> task) async {
+    final String debugName = task.constMeta.debugName;
+    final TimeWatch stopwatch = await interceptor.beforeExecuteNormal(debugName, task.hint);
+    final result = await super.executeNormal(task);
+    await interceptor.afterExecuteNormal(debugName, stopwatch);
```

Comment:
Same as before - we should loop the critical section and only report sparsely (report the average)

Reply:
That's a good exercise for a future contributor.

Reply:
(see replies above)